core/src/main/scala/io/chrisdavenport/curly/Main.scala

```diff
@@ -196,7 +194,8 @@ object Curly {
       P.until(sp).map(_.some) <* sp.? 
     } else P.unit.as(None)
     (code ~ body).map{
-      case (_, value) => Opt(optC.long, value)
+      case (_, Some(value)) => Opt(optC.long, value)
+      case (_, None) => Flag(optC.long)
```

Comment:
Looks like this could be just:
```scala
    (code *> body) match {
      case Some(value) => Opt(optC.long, value)
      case None => Flag(optC.long)
    }
```

Reply:
Addressed